pegen/pegen.c

```diff
@@ -53,7 +53,10 @@ raise_syntax_error(Parser *p, const char *errmsg, ...)
     if (!errstr) {
         goto error;
     }
-    if (p->tok->filename) {
+    if (p->tok->filename && PyUnicode_CompareWithASCIIString(p->tok->filename, "<string>") != 0) {
```

Comment:
I would like to avoid to call `PyUnicode_CompareWithASCIIString` and instead make the other branch work better.

Reply:
That'd be good, because we could rename the filename to `fstring` in `fstring_compile_expr`, like CPython does, more comfortably.